eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/mutable/UnmodifiableBag.java

```diff
@@ -438,4 +440,16 @@ protected Object readResolve()
     {
         return this.getMutableBag().selectUnique();
     }
+
+    @Override
+    public MutableSet<T> asSet()
+    {
+        return UnmodifiableMutableSet.of(this.getMutableBag().asSet());
```

Comment:
The current implementation will double wrap (two unmodifiable wrappers), no?

Reply:
@motlin yes maybe, but I thought it was safer like that. Maybe `UnmodifiableMutableSet.of` should check if it's not already an `UnmodifiableMutableSet`?